docs/src/doc/user-guide/advanced/kotlin-singleton.md

```diff
@@ -0,0 +1,17 @@
+Sometimes you need to store some Godot objects or references in a Kotlin singleton. This can cause some leaks when the program ends because you have to manually free objects and have the references collected. The `GodotStatic` interface lets you implement the function `collect` so you can write the appropriate cleanup code. Then you just need to call `registerAsSingleton` so it will be hooked to the Garbage Collector. 
+
+```kt
+object MySingleton : GodotStatic {
+    var ref: MyReference? = MyReference()
+    var obj: MyObject = MyObject()
+
+    init{
+        registerAsSingleton()
+    }
+
+    override fun collect(){
+        obj.free()
```

Comment:
Shouldn't be `queueFree()`? When I did only `free()` godot was not happy about it

Reply:
Not sure, @CedNaru wdyt? since you added this.